aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -271,8 +272,12 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
   } else {
     // Simulate cast to Complex on older MacOS by initializing real and imag parts
     if (dst_.is_complex() && !supportsComplex()) {
-      at::real(dst_) = src;
-      at::imag(dst_) = at::zeros_like(src);
+      if (!src.is_complex()) {
+        at::real(dst_) = src;
```

Comment:
This kind of assignment doesn't really do anything? do you mean `at::real(dst_).copy_(src)` ?

Reply:
Not sure I understand what do you mean. It copies (with cast) values of src into real part of `dst`, isn't it?

I.e. on MacOS13:
```
% python -c "import torch;print(torch.arange(10, device='mps').cfloat())"
tensor([0.+0.j, 1.+0.j, 2.+0.j, 3.+0.j, 4.+0.j, 5.+0.j, 6.+0.j, 7.+0.j, 8.+0.j, 9.+0.j],
       device='mps:0')
```
If it didn't work, I've assumed it would yield garbage, would't it?

Reply:
we have a special operator= on tensor that makes this work

Reply:
Changed it to use `copy_` as it's more explicit this way